### ISSUE TITLE
PathChangedINternally flawes

### ISSUE THREAD

**Issue by llehn** [gilbox/angular-flux-routing-example#2]
Hi there, exellent article and nice code. Yes I'm aware this is just a demo.

The implementation of PathChangedInternally doesn't really work (it does in your demo since you have no internal routing changes, since both routes first return false and then call API and then route which is then interpreted as not internal (!!payload.PathChangedInternal which is !!undefined). 

But if there is one internal route change, the app stop reacting to manual URL changes (esp. noticeable without html5-mode urls, ie. with #/ urls).

Conceptually, when the url changes, we'd need to know whether it's from some action which is currently in progress (i.e. triggered by code) or not. Currently the code looks at the last route change to find that out - which could be in the past and long done.

I'm not sure what the solution would be for that. Any ideas?

I made a small demo with this commit: https://github.com/llehn/angular-flux-routing-example/commit/699080ec5052494b2307eab3ed414faccb50b4a5 added a second link at teh top of the page, which doesn't do much but just navigate to /stuck. Then the back button doesn't work.

**Comment by gilbox** [comment]
@llehn thanks for the report. Even though I still think that the approach is solid, I have some regrets about posting the article as it is pretty confusing as well as this repo because of mistakes such as this (I suspect there are more mistakes). But I haven't had time to go back and fix everything.

Anyway, a `$watch`er was a poor choice for handling path changes. I've pushed up [a fix for this issue](https://github.com/gilbox/angular-flux-routing-example/commit/bf892ad27ee7c052c82e5a1a413937dceef7e24d). 

Please let me know if this fixes everything for you.

**Comment by llehn** [comment]
Thanks a lot! I had a very heavily modified codebase (basically took the idea you presented and added typescript to the team) so I couldn't apply that 100%. When I did it almost fixed everything, after I moved the state changing and location changing directly into the routestore it worked perfectly. 

Btw I didn't regret reading your articles at all, very very useful insights. There will be probably mor mistakes, but so has the code we devs ship and ask customers money for - so what. So keep up the writing and idea sharing - very highly appreciated.
